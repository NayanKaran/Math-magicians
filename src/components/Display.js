import React from 'react';
import PropTypes from 'prop-types';

function getInnerText(data) {
  const { total, operation, next } = data;
  if (!total && !operation && next) {
    return next;
  }
  if (total && !operation && !next) {
    return total;
  }
  if (total && operation && !next) {
    return `${total} ${operation}`;
  }
  if (total && operation && next) {
    return `${total} ${operation} ${next}`;
  }
  return '0';
}

const Display = (props) => {
  const { data, id } = props;
  return <p id={id}>{getInnerText(data)}</p>;
};

export default Display;

Display.propTypes = {
  data: PropTypes.shape({
    total: PropTypes.string,
    operation: PropTypes.string,
    next: PropTypes.string,
  }),
  id: PropTypes.string,
};

Display.defaultProps = {
  data: null,
  id: null,
};
