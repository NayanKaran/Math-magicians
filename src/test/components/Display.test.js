import renderer from 'react-test-renderer';
import Display from '../../components/Display';

describe('Testing Display component', () => {
  it('Testing with snapshot', () => {
    const data = {
      total: null,
      next: null,
      operation: null,
    };

    const tree = renderer.create(<Display data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
