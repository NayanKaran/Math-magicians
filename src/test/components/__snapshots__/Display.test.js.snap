// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Display component Testing with snapshot 1`] = `
<p
  id={null}
>
  0
</p>
`;
